query-engine/core/src/response_ir/internal.rs

```diff
@@ -479,18 +465,35 @@ fn serialize_objects(
     // to prevent expensive copying during serialization).
 
     // Finally, serialize the objects based on the selected fields.
-    let mut object_mapping = UncheckedItemsWithParents::with_capacity(result.scalars.records.len());
-    let db_field_names = result.scalars.field_names;
+    let mut object_mapping = UncheckedItemsWithParents::with_capacity(result.records.records.len());
+    let db_field_names = result.records.field_names;
     let model = result.model;
 
     let fields: Vec<_> = db_field_names
         .iter()
-        .filter_map(|f| model.fields().find_from_non_virtual_by_db_name(f).ok())
+        .map(|name| {
+            model
+                .fields()
+                .find_from_non_virtual_by_db_name(name)
+                .ok()
+                .and_then(|field| {
+                    typ.find_field(field.name())
+                        .map(|out_field| SerializedField::Model(field, out_field))
+                })
+                .or_else(|| {
+                    result
+                        .virtual_fields
+                        .iter()
+                        .find(|f| f.db_alias() == *name)
+                        .map(SerializedField::Virtual)
+                })
+                .expect("Field must be a known scalar or virtual")
+        })
```

Comment:
Is there a possibility of getting rid of the `.expect()`, and possibly any panic-inducing logic here?
Could you please expand the context a bit more to explain how it'd be possible to reach this panic state?

Reply:
> Could you please expand the context a bit more to explain how it'd be possible to reach this panic state?

Generally it shouldn't be except due to a bug in code, hence the panic. Basically this means that the query returned some unknown or unexpected fields.

Returning an error would certainly be possible here. I felt like a panic was more appropriate but I don't have a strong opinion in this case.

Reply:
There is a bunch of cases where unwrapping is fine when we know the error can't happen but the abstractions force us to acknowledge it. Dom always encouraged me to unwrap in the cases where we knew an error couldn't happen (eg: because validated earlier), but I'm not sure why it's preferable over an `expect`. If anything, you could add a very short comment that explains why it shouldn't be happening